packages/lodestar/src/network/interface.ts

```diff
@@ -12,12 +12,13 @@ import {
   RequestBody,
   ResponseBody, Status, SignedBeaconBlock,
 } from "@chainsafe/lodestar-types";
-
 import {Method, RequestId} from "../constants";
 import StrictEventEmitter from "strict-event-emitter-types";
 import {IGossip} from "./gossip/interface";
 
-export type ResponseCallbackFn = (response: {err?: Error; output?: ResponseBody}) => void;
+export type Response = {err?: Error; output: ResponseBody};
+
+export type ResponseCallbackFn = ((response: Partial<Response>) => void);
```

Comment:
Should't ResponseCallbackFn take a ResponseChunk?
It looks like responses are not yet given to the resp/req manager in a chunked format yet, but given as a full response body.

Reply:
I will change this in the morning, I don't think we need Response, BeaconBlocksByRangeResponse and BeaconBlocksByRootResponse

Reply:
and instead of `Partial<ResponseChunk>` we probably should make `output` in `ResponseChunk` optional